auth/api/groups/http/endpoint.go

```diff
@@ -4,44 +4,43 @@ import (
 	"context"
 
 	"github.com/go-kit/kit/endpoint"
-	"github.com/mainflux/mainflux/internal/groups"
+	"github.com/mainflux/mainflux/auth"
 	"github.com/mainflux/mainflux/pkg/errors"
 )
 
-func CreateGroupEndpoint(svc groups.Service) endpoint.Endpoint {
+func CreateGroupEndpoint(svc auth.Service) endpoint.Endpoint {
```

Comment:
Consider switching to the following package structure:
```
/auth/
    api/
        http/
            keys/
                doc.go
                requests.go
                responses.go
                endpoint.go
            groups/
                doc.go
                requests.go
                responses.go
                endpoint.go
            doc.go
            transport.go
```
Where `keys` and `groups` hold all the necessary (**unexported**) code for the respective entities. Only export one fabric method will accept mux object reference, and we'll end up with something like this in `auth/api/http/transport.go`:
```go

// MakeHandler returns a HTTP handler for API endpoints unifying Keys and Groups API.
func MakeHandler(svc auth.Service, tracer opentracing.Tracer) http.Handler {
	opts := []kithttp.ServerOption{
		kithttp.ServerErrorEncoder(encodeError),
	}
	mux := bone.New()
        keysAPI.New(mux)
        groupsAPI.New(mux)
	mux.GetFunc("/version", mainflux.Version("auth"))
	mux.Handle("/metrics", promhttp.Handler())
	return mux
}
```